builtin/providers/aws/resource_aws_instance.go

```diff
@@ -265,34 +265,68 @@ func resourceAwsInstanceCreate(d *schema.ResourceData, meta interface{}) error {
 	if v := d.Get("associate_public_ip_address"); v != nil {
 		associatePublicIPAddress = v.(bool)
 	}
+	subnet := false
+	var subnetID string
+	if v, ok := d.GetOk("subnet_id"); ok {
+		subnet = true
+		subnetID = v.(string)
+	}
```

Comment:
This block seems wrong-ish. I think could be worked into 296 below. Basically, we need to reproduce the request shaping that `goamz` does here: https://github.com/mitchellh/goamz/blob/master/ec2/ec2.go#L374